cvat/apps/engine/ddln/multiannotation.py

```diff
@@ -0,0 +1,159 @@
+import json
+import logging
+from pathlib import Path
+from tempfile import TemporaryDirectory
+from types import SimpleNamespace
+
+from django.db import transaction
+from rest_framework import serializers
+
+from cvat.apps.annotation.transports.csv import CsvDirectoryExporter
+from cvat.apps.annotation.transports.cvat import CVATImporter
+from cvat.apps.dataset_manager.util import make_zip_archive
+
+from cvat.apps.engine import models
+from cvat.apps.engine.ddln.sequences import extend_assignees
+from cvat.apps.engine.ddln.utils import (
+    write_task_mapping_file, write_ddln_yaml_file, get_sequence_id_mapping, get_annotation_request_id, guess_task_name
+)
+from cvat.apps.engine.models import Task, Segment
+from cvat.apps.engine.utils import natural_order
+from merge_annotations import merge_annotations
+
+logger = logging.getLogger(__name__)
+ignored_logger = logger.getChild("ignored")
+
+
+def request_extra_annotation(task, segments, assignees):
+    segments_data = [(s, s.length, set(s.get_performers())) for s in segments]
+
+    assignments, failed_segments = extend_assignees(segments_data, assignees)
+
+    if failed_segments:
+        raise FailedAssignmentError(failed_segments)
+
+    with transaction.atomic():
+        version = task.times_annotated
+        for segment, assignee in assignments:
+            db_job = models.Job()
+            db_job.segment = segment
+            db_job.version = version
+            db_job.assignee = assignee
+            db_job.save()
+        task.times_annotated += 1
+        task.save()
+
+
+class FailedAssignmentError(Exception):
+    """Error raised when proper assignees cannot be found for some of the segments"""
+    def __init__(self, failed_segments):
+        self.failed_segments = failed_segments
+
+
+def merge(task_id, file_path, acceptance_score):
+    task = Task.objects.get(pk=task_id)
+
+    with TemporaryDirectory() as root_dir:
+        root_dir = Path(root_dir)
+        versions_dir = root_dir / "Annotation_versions"
+        accepted_dir = root_dir / "Annotation_output"
+        rejected_dir = root_dir / "Rejected_annotation_output"
+        requires_more_dir = root_dir / "Requires_more_annotations"
+        logs_dir = root_dir / "Annotation_log"
+        for d in [versions_dir, accepted_dir, rejected_dir, requires_more_dir, logs_dir]:
+            d.mkdir()
+        log_file = logs_dir / "merge.log"
+        score_file = logs_dir / "scores.txt"
+
+        annotation_dirs = []
+        extra_annotation_dir = None
+        for version in range(task.times_annotated):
+            version_dir = versions_dir.joinpath("V{}".format(version + 1))
+            version_dir.mkdir()
+
+            _dump_version(task, version, version_dir)
+
+            is_extra_annotation = version == 3
```

Comment:
the rest of the logic allows for more than 4 versions, should it be `>= 3` here?

Reply:
The algorithm in `exp-devtools` is quite hard-corded for 3/4 versions. Supporting greater amount of versions requires fair amount of changes,  the comparison might be updated alongside the.